File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void preo(TreeNode* n,vector<int>& v)
    {
        if(n == NULL)
            return;
        v.push_back(n->val);
        preo(n->left,v);
        preo(n->right,v);
    }
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> ans;
        preo(root,ans);
        return ans;
    }
};

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> arr = new ArrayList<Integer>();
        preorder(root,arr);
        return arr;
        
    }
    private static void preorder(TreeNode n,List<Integer>arr){
        if(n == null)
            return;
    arr.add(n.val);
    preorder(n.left,arr);
    preorder(n.right,arr);}
}